x/mint/keeper/genesis.go

```diff
@@ -5,13 +5,13 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// InitGenesis new mint genesis
+// InitGenesis initializes the x/mint store with data from the genesis state.
 func (keeper Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, data *types.GenesisState) {
 	keeper.SetMinter(ctx, data.Minter)
 	ak.GetModuleAccount(ctx, types.ModuleName)
 }
```

Comment:
```suggestion
func (keeper Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, data *types.GenesisState) {
        genesisTime := ctx.BlockTime()
        // override the genesis time with the actual genesis time supplied in `InitChain`
        data.Minter.GenesisTime = &genesisTime
	keeper.SetMinter(ctx, data.Minter)
	ak.GetModuleAccount(ctx, types.ModuleName)
}
```
I would really prefer that we keep the genesis time identical. Since the block time here is set in `InitChain` which is the [genesis time provided by Tendermint](https://github.com/celestiaorg/celestia-core/blob/a1b07a1e6c77595466da9c61b37c83b4769b47e7/consensus/replay.go#L312), I suggest we do this. Ideally I would not have two genesis times in the genesis file to avoid confusion and just have it set here

Reply:
we could do that, but my impression of this PR was to stop overriding the minter's genesis state

if we move it to a different spot, is that even consensus breaking?

Reply:
> if we move it to a different spot, is that even consensus breaking?

I guess so since this would be using block time of height 0 instead of 1

Reply:
Can we modify mint's `GenesisState` so it doesn't include `GenesisTime`? 

> if we move it to a different spot, is that even consensus breaking?

Yes because time at height 1 is different to genesis time

Reply:
I'll try the approach @cmwaters proposed and (1) remove genesis time from the mint genesis state and (2) set it via `InitGenesis`.

I tried something similar to https://github.com/celestiaorg/celestia-app/pull/1578#discussion_r1196202725 before but couldn't get it to work.

Reply:
> Can we modify mint's GenesisState so it doesn't include GenesisTime?

we could but then I think we'd have to also implement extra protections

I think the other option is to do what this PR is doing, not overwrite, and then rely on either the user to copy and paste the value for an important genesis, or we find a way to pass the genesis time when setting the default genesis, which seems really doable imo

Less important as I don't think there are many instances of this, but if we ever wanted to regensis a testnet while also maintaining some minter state, then we couldn't do that if we keep overwriting

Reply:
cool, fortunatley whatever is chosen isn't very important. I think the most important thing is to merge #1987 and cut rc5 today

any improvements to the current minter are welcome, will defer @rootulp :slightly_smiling_face: , I'm full :shipit: